components/get_post_link_modal/get_post_link_modal.tsx

```diff
@@ -2,57 +2,69 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
-import PropTypes from 'prop-types';
 
 import ModalStore from 'stores/modal_store.jsx';
 import Constants from 'utils/constants';
 import * as Utils from 'utils/utils.jsx';
 import GetLinkModal from 'components/get_link_modal';
 
-export default class GetPostLinkModal extends React.PureComponent {
-    static propTypes = {
+type Props = {
+    currentTeamUrl: string;
+}
+
+type State = {
+    show: boolean;
+    postId: string;
+}
 
-        /**
-         * URL of current team
-         */
-        currentTeamUrl: PropTypes.string.isRequired,
-    };
+type Post = {
+    id: string;
+    message?: string;
+}
 
-    constructor(props) {
+export default class GetPostLinkModal extends React.PureComponent<Props, State> {
+    public constructor(props: Props) {
         super(props);
 
         this.state = {
             show: false,
-            post: {},
+            postId: 'undefined',
```

Comment:
This is an arbitrary value here the `undefined` string, probably makes more sense to have an optional `postId?` attribute in the state, and don't set it here, and check it before is used. Actually, I think it makes no sense to be able to show the link if you don't have the postId, so probably we can check it in the render and return `null` instead of the modal if the postId is not set.